### PR TITLE
[luci/pass] Refactor FuseAddWithTConvPass

### PR DESCRIPTION
**Pull request by jiwaszki**
This commit changes the order of searching for the pattern and adds tests for the pass.

ONE-DCO-1.0-Signed-off-by: Jan Iwaszkiewicz <j.iwaszkiewi@samsung.com>

### CODE REVIEW

**Comment by jiwaszki** [comment]
For: https://github.com/Samsung/ONE/issues/13685

@seanshpark some of my findings:

* As we discussed the checking of only one successor in https://github.com/Samsung/ONE/pull/13607#discussion_r1714920608 - it looks like the practice emerges in other passes as well. 
Here are the original lines from `fuse_add_with_tconv`:
https://github.com/Samsung/ONE/blob/987fafaa0b21023f8e6cb93aa9e8b586fc6a123a/compiler/luci/pass/src/FuseAddWithTConvPass.cpp#L62-L65
I have decided not to change this check as the task is to refactor order of the search.
* There were no tests in `FuseAddWithTConvPass.test.cpp` besides name checking, so I have added them.

**Comment by jiwaszki** [comment]
@seanshpark @jinevening PR is ready to review. I cannot add you to reviewers or change the label (is it blocked for contributors?).

File: compiler/luci/pass/src/FuseAddWithTConvPass.cpp [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2020 Samsung Electronics Co., Ltd. All Rights Reserved
+ * Copyright (c) 2020-2024 Samsung Electronics Co., Ltd. All Rights Reserved
```
Comment (seanshpark) [inline compiler/luci/pass/src/FuseAddWithTConvPass.cpp:2]: we don't need to do this change

File: compiler/luci/pass/src/FuseAddWithTConvPass.test.cpp [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2021 Samsung Electronics Co., Ltd. All Rights Reserved
+ * Copyright (c) 2021-2024 Samsung Electronics Co., Ltd. All Rights Reserved
```
Comment (seanshpark) [inline compiler/luci/pass/src/FuseAddWithTConvPass.test.cpp:2]: ditto

File: compiler/luci/pass/src/FuseAddWithTConvPass.cpp [hunk 3]
```diff
@@ -42,55 +49,39 @@ namespace
...
+  RETURN_FALSE_UNLESS(addition->rank() == 1);
+
   auto addition_dim = addition->dim(0).value();
   auto filter_channel_dim = filter->dim(0).value();
-  if (filter_channel_dim != addition_dim)
-    return false;
+  RETURN_FALSE_UNLESS(filter_channel_dim == addition_dim);
 
-  // fuse addition with transposed conv
+  // Fuse addition with transposed conv:
```
Comment (seanshpark) [inline compiler/luci/pass/src/FuseAddWithTConvPass.cpp:84]: Q) why add colon?

File: compiler/luci/pass/src/FuseAddWithTConvPass.cpp (hunk 1 above)
Comment (jiwaszki) [reply compiler/luci/pass/src/FuseAddWithTConvPass.cpp:2]: I wasn't sure about it as well... now I have guidance to follow. Will do!

File: compiler/luci/pass/src/FuseAddWithTConvPass.cpp (hunk 3 above)
Comment (jiwaszki) [reply compiler/luci/pass/src/FuseAddWithTConvPass.cpp:84]: It's just my personal style of writing comments. I can easily remove this.

Comment (seanshpark) [reply compiler/luci/pass/src/FuseAddWithTConvPass.cpp:84]: it's different from others but if you like to add `:`, ok to go :)

**Review by seanshpark — APPROVED** [review]
LGTM thank you!

**Pull request merged**